File: BAEKJOON/01978/solution.c
#include <stdio.h>
 
int is_prime(int n);
 
int main()
{
    int n;
    int i;
    int count = 0;
    int lim;
 
    scanf("%d", &lim);
 
    for (i = 0; i < lim; ++i)
    {
        scanf("%d", &n);
        if (is_prime(n)) { ++count; }
    }
    
    printf("%d\n", count);
    
    return 0;
}
 
int is_prime(int n)
{
    int i;
 
    if (n != 2)
    { 
        if ((n % 2) && (n != 1)) //Odd
        {
            for (i = 3; i*i <= n; i += 2)
            {
                if (n%i == 0)
                {
                    return 0;
                }
            }
        }
        else
        {
            return 0;
        }
    }
 
    return 1;
}

File: BAEKJOON/01978/solution.py
# https://www.acmicpc.net/problem/1978

def is_prime(x):
    if x == 1:
        return 0
    elif x == 2:
        return 1
    else:
        for i in range(2, x):
            if x%i == 0:
                return 0
    return 1


n = int(input())
count = 0
for s in input().split():
    count += is_prime(int(s))
print(count)
